docker/builder.ci.Dockerfile

```diff
@@ -0,0 +1,45 @@
+# syntax=docker/dockerfile:1
+
+# https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/
+# https://www.docker.com/blog/cross-compiling-rust-code-for-multiple-architectures/
+# https://www.docker.com/blog/multi-arch-build-and-images-the-simple-way/
+# https://github.com/cross-rs/cross/wiki/Recipes#openssl
+
+FROM --platform=$BUILDPLATFORM ubuntu:latest as builder
+
+ARG TARGETARCH
+
+RUN apt-get update && \
+  apt-get install -y build-essential clang cmake protobuf-compiler curl \
+  g++-aarch64-linux-gnu libc6-dev-arm64-cross \
```

Comment:
No, they aren't necessary. I also wondered if I should make the `aarch64` specific Rust toolset commands conditional. I thought maybe if they are always present then we can make better use of Docker caching, because at the cost of a few extra libraries we have a base layer that's good for both, rather than two almost identical layers that need to be cached separately. GH has a 10GB limit on cache size.

Reply:
I'll try to make these conditional and share the base cache layer